src/recordpage.cpp

```diff
@@ -801,9 +813,81 @@ QStringList RecordPage::applicationConfigurations()
 
 void RecordPage::appendOutput(const QString& text)
 {
+    // this regex finds ansi escapes and extracts all the numbers in it
+    QRegularExpression regex(QLatin1String("\\x1b\\[([0-9;]+)m"));
+
     QTextCursor cursor(ui->perfResultsTextEdit->document());
     cursor.movePosition(QTextCursor::End);
-    cursor.insertText(text);
+
+    auto ansiToHtml = [](const QStringList& numbers) -> QString {
+        static bool openColor = false;
+        // only foregroud colors are supported currently
+        const QHash<int, QColor> colors = {
```

Comment:
move to separate function that has a simple switch statement to do the mapping, instead of requiring a qhash construction _every time this lambda is called_!

Reply:
furthermore this won't work for systems using dark color schemes (which is what I use), we must not hard code the colors.

really, I think the proper solution here would be to figure out how to embed the konsole kpart to do this task for us, instead of trying to reinvent the wheel and suffering from all kinds of issues then

Reply:
As far as I know these colors are standardized

Reply:
they are not:
![Screenshot_20210810_101133](https://user-images.githubusercontent.com/57207/128832667-0397c949-5b4a-4165-987a-50c8788b4fe0.png)